lib/matplotlib/transforms.py

```diff
@@ -880,6 +879,60 @@ def update_from_path(self, path, ignore=None, updatex=True, updatey=True):
                 self._points[:, 1] = points[:, 1]
                 self._minpos[1] = minpos[1]
 
+    def _calc_extents_from_path(self, path, ignore, updatex=True, updatey=True):
+        """
+        Calculate the new bounds and minimum positive values for a `Bbox` from
+        the path.
+
+        Parameters
+        ----------
+        path : `~matplotlib.path.Path`
+        ignore : bool
+            - When ``True``, ignore the existing bounds of the `Bbox`.
+            - When ``False``, include the existing bounds of the `Bbox`.
+        updatex : bool
+            When ``True``, update the x-values.
+        updatey : bool
+            When ``True``, update the y-values.
+
+        Returns
+        -------
+        points : (2, 2) array
+        minpos : (2,) array
+        changed : bool
+        """
+        if ignore:
+            points = np.array([[np.inf, np.inf], [-np.inf, -np.inf]])
+            minpos = np.array([np.inf, np.inf])
+        else:
+            points = self._points.copy()
+            minpos = self._minpos.copy()
+
+        if updatex and updatey:
+            where = (np.isfinite(path.vertices[..., 0])
+                     & np.isfinite(path.vertices[..., 1]))
+        elif updatex:
+            where = np.isfinite(path.vertices[..., 0])
+        elif updatey:
+            where = np.isfinite(path.vertices[..., 1])
+        else:
+            return points, minpos, False
```

Comment:
If we are only updating "x", but the path has `(1, nan)` as the point, this will still include that position. Is that what we want? I think the original function removes this point from both the x and y calculations. So this could be simplified to just removing any row with a nan in it.

```suggestion
        if not (updatex or updatey):
            return points, minpos, False
        valid_verts = path.vertices[np.all(np.isfinite(path.vertices), axis=-1)]
```

Reply:
Great catch, you're right. Updated.